src/frontend/src/optimizer/plan_node/generic/scan.rs

```diff
@@ -247,7 +262,7 @@ impl Scan {
         Self::new_inner(
             table_name,
             output_col_idx,
-            table_desc,
```

Comment:
We can have everything reference `table_catalog`, but that requires a fair amount of refactoring. So do that in separate PR.